Source/ServerTrustEvaluation.swift

```diff
@@ -358,8 +412,53 @@ public extension SecTrust {
     var isValid: Bool {
         var result = SecTrustResultType.invalid
         let status = SecTrustEvaluate(self, &result)
+        
+        return (status.isSuccess) ? (result == .unspecified || result == .proceed) : false
+    }
+    
+    @discardableResult
+    func validate(policy: SecPolicy, errorProducer: (_ status: OSStatus, _ result: SecTrustResultType) -> Error) throws -> Bool {
+        return try apply(policy: policy).isValid(errorProducer: errorProducer)
+    }
+
+    @discardableResult
+    func isValid(errorProducer: (_ status: OSStatus, _ result: SecTrustResultType) -> Error) throws -> Bool {
```

Comment:
I'm fairly certain the `Bool` return value causes more harm than good here with these APIs. It would be worth a good think as to whether we really need them or not.

Reply:
Updated.